### PR TITLE
For PR approval of app (com.maplerain.telsis_translator_flutter)

### PR DESCRIPTION
**Pull request by vivian-ng**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

For approval please.

This is my first time submitting an app for review, apologies if I get things slightly wrong.

### CODE REVIEW

File: com.maplerain.telsis_translator_flutter.desktop [hunk 1]
```diff
@@ -0,0 +1,12 @@
+[Desktop Entry]
```
Comment (barthalion) [inline com.maplerain.telsis_translator_flutter.desktop:1]: You should maintain this file in your repo and distribute it in the tarball.

File: com.maplerain.telsis_translator_flutter.metainfo.xml [hunk 2]
```diff
@@ -0,0 +1,32 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (barthalion) [inline com.maplerain.telsis_translator_flutter.metainfo.xml:1]: You should maintain this file in your repo and distribute it in the tarball.

File: com.maplerain.telsis_translator_flutter.desktop [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+Version=1.0
+Type=Application
+
+Name=Telsis language translator
+Comment=App that translates to and from the Telsis language used in the Violet Evergarden series
+Categories=Utility;Amusement;
+
+Icon=com.maplerain.telsis_translator_flutter
```
Comment (barthalion) [inline com.maplerain.telsis_translator_flutter.desktop:9]: Same about the icon file.

File: com.maplerain.telsis_translator_flutter.yml [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+  # telsis_translator_flutter
+  - name: telsis_translator_flutter
+    buildsystem: simple
+    only-arches:
+      - x86_64
+    build-commands:
+      - "./build-flatpak.sh"
+    sources:
+      - type: file
+        url: https://github.com/vivian-ng/telsis_translator_flutter/releases/download/0.2.1/telsis-translator-flutter_0.2.1.tar.gz
```
Comment (barthalion) [inline com.maplerain.telsis_translator_flutter.yml:26]: As I assume this is a pre-built binary, you should limit it to x86_64: https://github.com/flathub/flathub/wiki/App-Maintenance#only-build-on-x86_64

File: com.maplerain.telsis_translator_flutter.desktop (hunk 1 above)
Comment (vivian-ng) [reply com.maplerain.telsis_translator_flutter.desktop:1]: Yes, it is in the tarball. I saw this file on other repos and blindly followed. My apologies. 🙇‍♀️

File: com.maplerain.telsis_translator_flutter.metainfo.xml (hunk 2 above)
Comment (vivian-ng) [reply com.maplerain.telsis_translator_flutter.metainfo.xml:1]: Yes, it is in the tarball. I saw this file on other repos and blindly followed. My apologies. 🙇‍♀️

File: com.maplerain.telsis_translator_flutter.desktop (hunk 3 above)
Comment (vivian-ng) [reply com.maplerain.telsis_translator_flutter.desktop:9]: Yes, it is in the tarball. I saw this file on other repos and blindly followed. My apologies. 🙇‍♀️

File: com.maplerain.telsis_translator_flutter.yml (hunk 4 above)
Comment (vivian-ng) [reply com.maplerain.telsis_translator_flutter.yml:26]: I have deleted the desktop, metainfo, and icon files. Plus added the `flathub.json` file to limit to x86_64.

File: com.maplerain.telsis_translator_flutter.yml [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+  - --device=dri
+  - --share=network
+modules:
+  # telsis_translator_flutter
+  - name: telsis_translator_flutter
+    buildsystem: simple
+    only-arches:
+      - x86_64
+    build-commands:
+      - "./build-flatpak.sh"
```
Comment (hfiguiere) [inline com.maplerain.telsis_translator_flutter.yml:23]: you don't need that. It's unmaintainable.

File: com.maplerain.telsis_translator_flutter.yml [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+    sources:
+      - type: file
+        url: https://github.com/vivian-ng/telsis_translator_flutter/releases/download/0.2.1/telsis-translator-flutter_0.2.1.tar.gz
+        sha256: 118a8ff29226a041cbacceb71bb95a21c8d1ba90c020a5ee1ef3ec8c9f4f0746
+      - type: dir
```
Comment (hfiguiere) [inline com.maplerain.telsis_translator_flutter.yml:28]: this is incorrect. There is no such directory.

Comment (hfiguiere) [reply com.maplerain.telsis_translator_flutter.yml:28]: or more like it's `.` which in the end goes back to what I as saying. it's incorrect.

File: com.maplerain.telsis_translator_flutter.yml [hunk 7]
```diff
@@ -0,0 +1,34 @@
+# yaml-language-server: $schema=https://raw.githubusercontent.com/flatpak/flatpak-builder/main/data/flatpak-manifest.schema.json
+
+---
+app-id: com.maplerain.telsis_translator_flutter
```
Comment (hfiguiere) [inline com.maplerain.telsis_translator_flutter.yml:4]: `maplerain.com` doesn't exist on wherever it is being hosted. Is this domain really controlled by the application developers?

File: com.maplerain.telsis_translator_flutter.yml [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+modules:
+  # telsis_translator_flutter
+  - name: telsis_translator_flutter
+    buildsystem: simple
+    only-arches:
+      - x86_64
+    build-commands:
+      - "./build-flatpak.sh"
+    sources:
+      - type: file
```
Comment (hfiguiere) [inline com.maplerain.telsis_translator_flutter.yml:25]: if you set this to `type: archive` it is automatically extracted.

File: com.maplerain.telsis_translator_flutter.yml (hunk 6 above)
Comment (vivian-ng) [reply com.maplerain.telsis_translator_flutter.yml]: I have removed the dir section.

File: com.maplerain.telsis_translator_flutter.yml (hunk 7 above)
Comment (vivian-ng) [reply com.maplerain.telsis_translator_flutter.yml:4]: I own the domain name "maplerain.com" and my main business site is at [www.maplerain.com](https://www.maplerain.com).

File: com.maplerain.telsis_translator_flutter.yml (hunk 8 above)
Comment (vivian-ng) [reply com.maplerain.telsis_translator_flutter.yml:25]: I was following an example I found for another flutter-based app on how to package it.
Specifically, this one [here](https://github.com/flathub/codes.merritt.Nyrna).
Will it work okay? Or should I find a way to put all the build commands from the script into the main YAML file?

File: com.maplerain.telsis_translator_flutter.yml (hunk 5 above)
Comment (vivian-ng) [reply com.maplerain.telsis_translator_flutter.yml:23]: Sorry, I am not sure I fully understand. Do you mean the script file is unmaintainable? Or the only_arches part? Any hints on how to resolve this? It is my first time trying to package for Flathub and while I could use flatpak-builder to locally export to a flatpak, I understand that the automated build process on Flathub may be different and what worked locally for me may not work here.

File: com.maplerain.telsis_translator_flutter.yml (hunk 8 above)
Comment (hfiguiere) [reply com.maplerain.telsis_translator_flutter.yml:25]: it was done post review. I would have had the same comments. This script is excessiely complex for no reason. It's max 5 lines of `build-commands`

File: com.maplerain.telsis_translator_flutter.yml (hunk 5 above)
Comment (hfiguiere) [reply com.maplerain.telsis_translator_flutter.yml:23]: This script is excessiely complex for no reason. It's max 5 lines of build-commands

File: com.maplerain.telsis_translator_flutter.yml (hunk 8 above)
Comment (vivian-ng) [reply com.maplerain.telsis_translator_flutter.yml:25]: Okay, I kind of understand what you mean. I will use [this](https://github.com/flathub/com.github.tglima.Myfluttercrypto) as an example to amend the YAML file and get rid of the build script.

File: com.maplerain.telsis_translator_flutter.yml [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+      - chmod +x telsis_translator_flutter
+      - mv com.maplerain.telsis_translator_flutter.png /app/share/icons/hicolor/256x256/apps/
+      - mv com.maplerain.telsis_translator_flutter.desktop /app/share/applications/
+      - mv com.maplerain.telsis_translator_flutter.metainfo.xml /app/share/metainfo/
+      - mv * /app/
+      - mkdir -p /app/bin
+      - ln -s /app/telsis_translator_flutter /app/bin/telsis_translator_flutter
+    sources:
+      - type: archive
+        url: https://github.com/vivian-ng/telsis_translator_flutter/releases/download/0.2.1/telsis-translator-flutter_0.2.1.tar.gz
```
Comment (hfiguiere) [inline com.maplerain.telsis_translator_flutter.yml:32]:
```suggestion
        url: https://github.com/vivian-ng/telsis_translator_flutter/releases/download/0.2.1/telsis-translator-flutter_0.2.1.tar.gz
        strip-components: 0
```

This is needed given the layout of the archive.

File: com.maplerain.telsis_translator_flutter.yml (hunk 5 above)
Comment (vivian-ng) [reply com.maplerain.telsis_translator_flutter.yml]: I have deleted the script file.

File: com.maplerain.telsis_translator_flutter.yml [hunk 10]
```diff
@@ -0,0 +1,38 @@
...
+    only-arches:
+      - x86_64
+    build-commands:
+      - chmod +x telsis_translator_flutter
+      - mv com.maplerain.telsis_translator_flutter.png /app/share/icons/hicolor/256x256/apps/
+      - mv com.maplerain.telsis_translator_flutter.desktop /app/share/applications/
+      - mv com.maplerain.telsis_translator_flutter.metainfo.xml /app/share/metainfo/
+      - mv * /app/
+      - mkdir -p /app/bin
+      - ln -s /app/telsis_translator_flutter /app/bin/telsis_translator_flutter
```
Comment (hfiguiere) [inline com.maplerain.telsis_translator_flutter.yml:29]:
```suggestion
      - install -Dm644 com.maplerain.telsis_translator_flutter.png /app/share/icons/hicolor/256x256/apps/
      - install -Dm644 com.maplerain.telsis_translator_flutter.desktop /app/share/applications/
      - install -Dm644 com.maplerain.telsis_translator_flutter.metainfo.xml /app/share/metainfo/
      - mv data lib telsis_translator_flutter /app/
      - mkdir -p /app/bin
      - ln -s /app/telsis_translator_flutter /app/bin/telsis_translator_flutter
```

ok, that's 6 lines.

File: com.maplerain.telsis_translator_flutter.yml [hunk 11]
```diff
@@ -0,0 +1,38 @@
...
+  - --socket=fallback-x11
+  - --socket=wayland
+  - --device=dri
+  - --share=network
+modules:
+  # telsis_translator_flutter
+  - name: telsis_translator_flutter
+    buildsystem: simple
+    only-arches:
+      - x86_64
```
Comment (hfiguiere) [inline com.maplerain.telsis_translator_flutter.yml:21]:
```suggestion
```

This is not needed.

File: com.maplerain.telsis_translator_flutter.yml [hunk 12]
```diff
@@ -0,0 +1,38 @@
...
+      - ln -s /app/telsis_translator_flutter /app/bin/telsis_translator_flutter
+    sources:
+      - type: archive
+        url: https://github.com/vivian-ng/telsis_translator_flutter/releases/download/0.2.1/telsis-translator-flutter_0.2.1.tar.gz
+        sha256: 128b6b2f48ef6706fb159a1ed44939ca1c401085e3b544924e7da6668740b338
+      # - type: git
+      #   url: https://github.com/vivian-ng/telsis_translator_flutter.git
+      #   branch: main
+      # - type: file
+      #   path: build-flatpak.sh
```
Comment (hfiguiere) [inline com.maplerain.telsis_translator_flutter.yml:38]:
```suggestion
```

File: com.maplerain.telsis_translator_flutter.yml (hunk 11 above)
Comment (vivian-ng) [reply com.maplerain.telsis_translator_flutter.yml]: Okay, thank you for pointing this out.

File: com.maplerain.telsis_translator_flutter.yml (hunk 12 above)
Comment (vivian-ng) [reply com.maplerain.telsis_translator_flutter.yml]: Removed the useless comments.

File: com.maplerain.telsis_translator_flutter.yml (hunk 10 above)
Comment (vivian-ng) [reply com.maplerain.telsis_translator_flutter.yml]: Thank you for the suggestion. I have changed the archive to remove the `packaging\linux` directory (also updated the sha256sum) and made the relevant changes. Please take a look again.

**Comment by hfiguiere** [comment]
bot, build com.maplerain.telsis_translator_flutter

File: com.maplerain.telsis_translator_flutter.yml [hunk 13]
```diff
@@ -0,0 +1,32 @@
...
+  - --device=dri
+  - --share=network
+modules:
+  # telsis_translator_flutter
+  - name: telsis_translator_flutter
+    buildsystem: simple
+    build-commands:
+      - install -Dm644 com.maplerain.telsis_translator_flutter.png /app/share/icons/hicolor/256x256/apps/
+      - install -Dm644 com.maplerain.telsis_translator_flutter.desktop /app/share/applications/
+      - install -Dm644 com.maplerain.telsis_translator_flutter.metainfo.xml /app/share/metainfo/
```
Comment (hfiguiere) [inline com.maplerain.telsis_translator_flutter.yml:23]:
```suggestion
      - install -Dm644 com.maplerain.telsis_translator_flutter.png -t /app/share/icons/hicolor/256x256/apps/
      - install -Dm644 com.maplerain.telsis_translator_flutter.desktop -t /app/share/applications/
      - install -Dm644 com.maplerain.telsis_translator_flutter.metainfo.xml -t /app/share/metainfo/
```

File: com.maplerain.telsis_translator_flutter.yml [hunk 14]
```diff
@@ -0,0 +1,32 @@
...
+  - --device=dri
+  - --share=network
+modules:
+  # telsis_translator_flutter
+  - name: telsis_translator_flutter
+    buildsystem: simple
+    build-commands:
+      - mv com.maplerain.telsis_translator_flutter.png /app/share/icons/hicolor/256x256/apps/
+      - mv com.maplerain.telsis_translator_flutter.desktop /app/share/applications/
+      - mv com.maplerain.telsis_translator_flutter.metainfo.xml /app/share/metainfo/
```
Comment (hfiguiere) [inline com.maplerain.telsis_translator_flutter.yml:23]:
```suggestion
      - install -Dm644 com.maplerain.telsis_translator_flutter.png -t /app/share/icons/hicolor/256x256/apps/
      - install -Dm644 com.maplerain.telsis_translator_flutter.desktop -t /app/share/applications/
      - install -Dm644 com.maplerain.telsis_translator_flutter.metainfo.xml -t /app/share/metainfo/
```

Comment (vivian-ng) [reply com.maplerain.telsis_translator_flutter.yml]: I am stupid for not understanding the `install` command.

**Comment by hfiguiere** [comment]
bot, build com.maplerain.telsis_translator_flutter

**Comment by hfiguiere** [comment]
```
builddir/files/share/appdata/com.maplerain.telsis_translator_flutter.appdata.xml: FAILED:
• file-invalid          : <screenshot> failed to load [https://github.com/vivian-ng/telsis_translator_flutter/blob/main/screenshots/onLinux.png]
Validation of files failed
```

**Comment by vivian-ng** [comment]
I changed to use a `raw.githubusercontent.com` link for the image in the XML file and repackaged the tarball. The sha256sum has been updated in the YAML file.

**Comment by hfiguiere** [comment]
bot, build com.maplerain.telsis_translator_flutter

**Comment by vivian-ng** [comment]
Other than the warning
```
Warning: Not exporting file com.maplerain.telsis_translator_flutter.appdata.xml of unsupported type.
```
I was able to install and test the build from the testing repository.

**Pull request closed without merging**